src/main/java/org/prebid/server/bidder/connectAd/ConnectAdBidder.java

```diff
@@ -0,0 +1,190 @@
+import java.util.stream.Collectors;
+
+public class ConnectAdBidder implements Bidder<BidRequest> {
+
+    private static final TypeReference<ExtPrebid<?, ExtImpConnectAd>> CONNECTAD_EXT_TYPE_REFERENCE =
+            new TypeReference<ExtPrebid<?, ExtImpConnectAd>>() {
+            };
+
+    private final String endpointUrl;
+    private final JacksonMapper mapper;
+
+    public ConnectAdBidder(String endpointUrl, JacksonMapper mapper) {
+        this.endpointUrl = HttpUtil.validateUrl(Objects.requireNonNull(endpointUrl));
+        this.mapper = Objects.requireNonNull(mapper);
+    }
+
+    @Override
+    public Result<List<HttpRequest<BidRequest>>> makeHttpRequests(BidRequest request) {
+        final String page = request.getSite() != null ? request.getSite().getPage() : null;
+        final int secure = StringUtils.isNotBlank(page) && page.contains("https") ? 1 : 0;
+
+        final List<BidderError> errors = new ArrayList<>();
+        final List<Imp> processedImps = new ArrayList<>();
+
+        for (Imp imp : request.getImp()) {
+            try {
+                final ExtImpConnectAd impExt = parseImpExt(imp);
+                final Imp updatedImp = updateImp(imp, secure, impExt);
+                processedImps.add(updatedImp);
+            } catch (PreBidException e) {
+                errors.add(BidderError.badInput(e.getMessage()));
+            }
+        }
+        if (CollectionUtils.isNotEmpty(errors)) {
+            errors.add(BidderError.badInput("Error in preprocess of Imp"));
+            return Result.of(Collections.emptyList(), errors);
+        }
+        final BidRequest outgoingRequest = request.toBuilder().imp(processedImps).build();
+
+        final String body = mapper.encode(outgoingRequest);
+
+        final MultiMap headers = HttpUtil.headers();
+        updateHeadersDueToRequest(headers, request);
+
+        return Result.of(Collections.singletonList(
+                HttpRequest.<BidRequest>builder()
+                        .method(HttpMethod.POST)
+                        .uri(endpointUrl)
+                        .headers(headers)
+                        .payload(outgoingRequest)
+                        .body(body)
+                        .build()),
+                errors);
+    }
+
+    private ExtImpConnectAd parseImpExt(Imp imp) {
+        final ExtImpConnectAd extImpConnectAd;
+        try {
+            extImpConnectAd = mapper.mapper().convertValue(imp.getExt(), CONNECTAD_EXT_TYPE_REFERENCE).getBidder();
+        } catch (IllegalArgumentException e) {
+            throw new PreBidException(String.format("Impression id=%s, has invalid Ext", imp.getId()));
+        }
+
+        if (extImpConnectAd.getSiteId().equals(NumberUtils.INTEGER_ZERO)) {
+            throw new PreBidException(String.format("Impression id=%s, has no siteId present", imp.getId()));
+        }
+        return extImpConnectAd;
+    }
+
+    private Imp updateImp(Imp imp, Integer secure, ExtImpConnectAd extImp) {
```

Comment:
May be it make sense to pass here only needed parameters instead of whole `ExtImpConnectAd`.